File: LeetCode-2023/2315.统计星号.cpp
/*
 * @lc app=leetcode.cn id=2315 lang=cpp
 *
 * [2315] 统计星号
 */

// @lc code=start
class Solution {
public:
    int countAsterisks(string s) {
        bool flag = true;
        int count = 0;
        for (int i = 0; i < s.size(); i++) {
            if(s[i] == '|') {
                flag = !flag;
            }
            if(flag == true && s[i] == '*') {
                count++;
            }
        }
        return count;
    }
};
// @lc code=end

File: LeetCode-2023/2315.统计星号.py
#
# @lc app=leetcode.cn id=2315 lang=python3
#
# [2315] 统计星号
#

# @lc code=start
class Solution:
    def countAsterisks(self, s: str) -> int:
        flag = True
        count = 0
        for c in s:
            if c == '|':
                flag = not flag
            if flag == True and c == '*':
                count += 1
        return count
# @lc code=end

